app/controllers/player_controller.rb

```diff
@@ -0,0 +1,13 @@
+class PlayerController < ApplicationController
+  #history of 1 game
+  def leaderboard
+    type = params[:type]
+    team_size = params[:team_size]
+    return render json: {error: "game name required"}, status: :bas_request if type == nil
```

Comment:
It is [Ruby convention](https://github.com/rubocop-hq/ruby-style-guide#predicate-methods) to use `.nil?` instead of explicitly comparing using `==`.
Also, you have a type in the status. I think you meant `:bad_request`
```suggestion
    return render json: {error: "game type required"}, status: :bad_request if type.nil?
```